contracts/utils/contracts/src/LibERC20Token.sol

```diff
@@ -0,0 +1,116 @@
+    ///      Reverts if `false` is returned or if the return
+    ///      data length is nonzero and not 32 bytes.
+    /// @param token The address of the token contract.
+    /// @param spender The address that receives an allowance.
+    /// @param allowance The allowance to set.
+    function approve(
+        address token,
+        address spender,
+        uint256 allowance
+    )
+        internal
+    {
+        bytes memory callData = abi.encodeWithSelector(
+            0x095ea7b3,
+            spender,
+            allowance
+        );
+        _callWithOptionalBooleanResult(token, callData);
+    }
+
+    /// @dev Calls `IERC20Token(token).transfer()`.
+    ///      Reverts if `false` is returned or if the return
+    ///      data length is nonzero and not 32 bytes.
+    /// @param token The address of the token contract.
+    /// @param to The address that receives the tokens
+    /// @param amount Number of tokens to transfer.
+    function transfer(
+        address token,
+        address to,
+        uint256 amount
+    )
+        internal
+    {
+        bytes memory callData = abi.encodeWithSelector(
+            0xa9059cbb,
+            to,
+            amount
+        );
+        _callWithOptionalBooleanResult(token, callData);
+    }
+
+    /// @dev Calls `IERC20Token(token).transferFrom()`.
+    ///      Reverts if `false` is returned or if the return
+    ///      data length is nonzero and not 32 bytes.
+    /// @param token The address of the token contract.
+    /// @param from The owner of the tokens.
+    /// @param to The address that receives the tokens
+    /// @param amount Number of tokens to transfer.
+    function transferFrom(
+        address token,
+        address from,
+        address to,
+        uint256 amount
+    )
+        internal
+    {
+        bytes memory callData = abi.encodeWithSelector(
+            0x23b872dd,
+            from,
+            to,
+            amount
+        );
+        _callWithOptionalBooleanResult(token, callData);
+    }
+
+    /// @dev Executes a call on address `target` with calldata `callData`
+    ///      and asserts that either nothing was returned or a single boolean
+    ///      was returned equal to `true`.
+    /// @param target The call target.
+    /// @param callData The abi-encoded call data.
+    function _callWithOptionalBooleanResult(
+        address target,
+        bytes memory callData
+    )
+        private
+    {
+        (bool didSucceed, bytes memory resultData) = target.call(callData);
+        if (didSucceed) {
+            if (resultData.length == 0) {
+                return;
+            }
+            if (resultData.length == 32) {
+                uint256 result;
+                assembly { result := mload(add(resultData, 0x20)) }
+                if (result == 1) {
+                    return;
+                }
+            }
+        }
+        assembly { revert(add(resultData, 0x20), mload(resultData)) }
```

Comment:
We use this pattern in a lot of different places too. Maybe we should use `LibRichErrors.rrevert`?